src/main/java/fi/metatavu/metaform/server/rest/MetaformsApiImpl.java

```diff
@@ -296,11 +313,48 @@ public Response listReplies(UUID metaformId, UUID userId, String createdBeforePa
         modifiedAfter,
         includeRevisions != null && includeRevisions,
         fieldFilters);
-    
+    replies.forEach(reply -> auditLogEntryController.createAuditLogEntry(getLoggerUserId(), AuditLogEntryType.VIEW_REPLY,
+            reply.getId(), null, reply.getId()+" was viewed by user "+getLoggerUserId()));
     List<Reply> result = getPermittedReplies(replies, AuthorizationScope.REPLY_VIEW).stream()
       .map(entity -> replyTranslator.translateReply(metaformEntity, entity, null))
       .collect(Collectors.toList());
-    
+
+    return createOk(result);
+  }
+
+  @Override
+  public Response findAuditLogEntries(UUID metaformId, UUID userId, String createdBeforeParam, String createdAfterParam, String modifiedBeforeParam, String modifiedAfterParam) {
+    if (!hasRealmRole(VIEW_AUDIT_LOGS_ROLE)) {
+      return createForbidden("Only users with "+VIEW_AUDIT_LOGS_ROLE+" can access this view");
+    }
+    OffsetDateTime createdBefore = parseTime(createdBeforeParam);
+    OffsetDateTime createdAfter = parseTime(createdAfterParam);
+    OffsetDateTime modifiedBefore = parseTime(modifiedBeforeParam);
+    OffsetDateTime modifiedAfter = parseTime(modifiedAfterParam);
+
+    if (!isRealmUser()) {
+      throw new ForbiddenException(ANONYMOUS_USERS_MESSAGE);
+    }
+
+    fi.metatavu.metaform.server.persistence.model.Metaform metaform = metaformController.findMetaformById(metaformId);
+    if (metaform == null) {
+      throw new NotFoundException(NOT_FOUND_MESSAGE);
+    }
+
+    List<fi.metatavu.metaform.server.persistence.model.Reply> replies = replyController.listReplies(metaform,
```

Comment:
Remove these replies, it would be impossible to filter e.g. deleted replies, if you filter by replies found from the form